tools/CheckMetadata.py

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env python
+
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+import datetime
+import git
+import os
+import re
+import unicodedata
+import unittest
+import urllib
+import yaml
+
+VERSION_PATTERN = r'(\d{4})\.(\d{2})\.(\d{2})(\.\d+)?'
+DOI_PATTERN = r'10\.\d{4,9}/zenodo\.(\d+)'
+ORCID_PATTERN = r'\d{4}-\d{4}-\d{4}-\d{3}[0-9X]'
+
+
+def simplify_name(name):
+    """Converts the `name` to lower-case ASCII for fuzzy comparisons."""
+    return unicodedata.normalize('NFKD',
+                                 name.lower()).encode('ascii', 'ignore')
+
+
+class TestMetadata(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        # Always work with the repository that contains this file
+        cls.repo = git.Repo(__file__, search_parent_directories=True)
+        with open(os.path.join(cls.repo.working_dir, 'Metadata.yaml'),
```

Comment:
should there be some nice error message if this fails? E.g. because we are running from Formaline (or a github version), or we are not running from within the source tree at all?

Reply:
It raises a Py exception, so it already prints a relatively nice error message:

```
Traceback (most recent call last):
  File "tools/CheckMetadata.py", line 31, in setUpClass
    cls.repo = git.Repo(__file__, search_parent_directories=False)
  File "/Users/nlf/.virtualenvs/spectre/lib/python3.8/site-packages/git/repo/base.py", line 181, in __init__
    raise InvalidGitRepositoryError(epath)
git.exc.InvalidGitRepositoryError: tools/CheckMetadata.py
```

Added a bit more info to the error message now though.

Reply:
okay. Is there ever any need to run this on a release or formaline output? If so we could recursively search up directories for a `Metadata.yaml` if we can't find a git repo

Reply:
I can't think of a reason this needs to run outside the repo